kernel/src/engine/default/storage.rs

```diff
@@ -0,0 +1,29 @@
+#[cfg(feature = "cloud")]
+use hdfs_native_object_store::HdfsObjectStore;
+use object_store::parse_url_opts as object_store_parse_url_opts;
+use object_store::path::Path;
+use object_store::{Error, ObjectStore};
+use url::Url;
+
+pub fn parse_url_opts<I, K, V>(url: &Url, options: I) -> Result<(Box<dyn ObjectStore>, Path), Error>
+where
+    I: IntoIterator<Item = (K, V)>,
+    K: AsRef<str>,
+    V: Into<String>,
+{
+    #[cfg(feature = "cloud")]
```

Comment:
Can we pull this out into it's own function that is only enabled with the feature? Freestanding like this might be confusing.

Reply:
Updated, is this what you meant?